community/contributing/bisect_workflow.rst

```diff
@@ -0,0 +1,131 @@
+process, you can now start bisecting the regression. The Git version control
+system offers a built-in command for this: ``git bisect``. This makes the
+process semi-automated as you only have to build the engine, run it and try to
+reproduce the bug.
+
+.. note::
+
+    Before bisecting a regression, you need to set up a build environment to
+    compile Godot from source. To do so, read the
+    :ref:`Compiling <toc-devel-compiling>` page for your target platform.
+    (Compiling Godot from source doesn't require C++ programming knowledge.)
+
+To start bisecting, you must first determine the commit hashes (identifiers) of
+the "bad" and "good" build. "bad" refers to the build that exhibits the bug,
+whereas "good" refers to the version that doesn't exhibit the bug. If you're
+using a pre-release build as the "good" or "bad" build, browse the `download
+mirror <https://downloads.tuxfamily.org/godotengine/>`__, go to the folder that
+contains the pre-release you downloaded and look for the ``README.txt`` file.
+The commit hash is written inside that file.
+
+If you're using a stable release as the "good" or "bad" build, use one of the
+following commit hashes depending on the version:
+
+.. code-block:: none
+
+    3.1.2: 0587df4aa5f2977350cc80b1522cdc1e483c4515
+    3.1.1: 66baa3b633fe904ea0d90a9688d602d9f3a0b3bd
+    3.1: 320f49f204cfbf9b480fe62aaa7718afb74920a5
+    3.0.6: 8ac39d886307d76c286e804e027fc39f6b5aaac6
+    3.0.5: 6a88e22423b5212fdb660d3cacad362382fbb805
+    3.0.4: b116a45f07586d3765a3f39c77c3632d0fc588cb
+    3.0.3: 25047050b7798efd976cb591d2868fed04f2c895
+    3.0.2: 2ff3b08583766c975d76d8c39ba0937a5c9bf064
+    3.0.1: 8ea709a7a93dab01b0e33df168cba680ef016bf0
+    3.0: 9bd402698cfa299b79b9144f8d7744c308a4e085
+
+:ref:`Get Godot's source code using Git <doc_getting_source>`. Once this
+is done, in the terminal window, use ``cd`` to reach the Godot repository
+folder and enter the following command:
+
+.. code-block::
+
+    # <good> is the commit hash of the build that works as expected.
+    # <bad> is the commit hash of the build exhibiting the bug.
+    git bisect start <good> <bad>
+
+Compile Godot. This assumes you've set up a build environment:
+
+.. code-block::
+
+    # <platform> is the platform you're targeting for regression testing,
+    # like "windows", "x11" or "osx".
+    scons platform=<platform> -j4
+
+Since building Godot takes a while, you want to dedicate as many CPU threads as
+possible to the task. This is what the ``-j`` parameter does. Here, the command
+assigns 4 CPU threads to compiling Godot.
+
+Run the binary located in the ``bin/`` folder and try to reproduce the bug.
+
+If the build **still** exhibits the bug, run the following command:
+
+.. code-block::
+
+    git bisect bad
+
+If the build **does not** exhibit the bug, run the following command:
+
+.. code-block::
+
+    git bisect good
+
+You need to repeat this operation several times. The longer the commit range,
```

Comment:
Clarify "this operation" to mean building, testing, and then issuing a new bisect command.